src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -991,6 +991,7 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 						this._connectionStatusManager.deleteConnection(uri);
 						resolve({ connected: connectResult, errorMessage: errorMessage, errorCode: errorCode, callStack: callStack, connectionProfile: connection });
 					} else {
+						connection.options.isCloud = connectionMngInfo.serverInfo.isCloud;
```

Comment:
Might make more sense to utilize command 'when' statements and use the context mssql extension is already setting for isCloud

Reply:
Can you point me to an example to learn from for this?  This isn't to decide to show menus or not, but rather if a successfully connected to server should be added to the CMS registration list.  The other check was already there was duplicate name.  Maybe there is a better place I could do this check all together?

Reply:
Checked with Karl and when has more to do with what commands are valid for a given connection context.  This is a validation check for adding an item to a viewlet so keeping it this way for now.

Reply:
I think this code path is for validation of dialog box input.  I'm not sure how we can use the when clause for this task.  Wouldn't we need a new concept like "Validation" or "Server" or something to attach the when to, since there isn't an applicable command in this case.